### ISSUE TITLE
Solaar does not maintain settings after device sleeps

### ISSUE THREAD

**Issue by krrk** [pwr-Solaar/Solaar#443]
The settings applied in Solaar only work when the program is first opened or when making a change. After a short while the settings are lost. For my wireless mouse Logitech MX Master it appears that after not using the mouse for some time it must go into some power save mode. When it is awoken it will have reset to its default settings (1000 DPI and No Smooth Scrolling). Then I have to either close and open Solaar or change the DPI to something other than 1600 (the maximum) and then change it back and also deactivate and reactivate smooth scrolling. The command line interface also shows that the mouse has reset to its default settings.

Is there supposed to be some sort of thread watching the mouse to re-apply changes when necessary built into Solaar? If so it is not working for me. Otherwise please treat this issue as a feature request for such a daemon. Solaar is practically not worth the effort for me as it stands. After correctly setting libinput for MOUSE_DPI and MOUSE_WHEEL_CLICK_COUNT for 1600 and smooth scroling when the mouse resets to default its slower and basically unable to scroll.

I have installed Solaar with Python3 using the latest git version.

**Comment by Zefiren** [comment]
Can confirm I have the same issue with my Logitech K400 Plus keyboard as it resets back to default settings (fn-swap=true) after a while with no usage. This was the only thing I installed Solaar for to fix, so it is rather useless for it to forget the settings after a while. 

If I have some spare time, I'm interested if I would be able to fix this sleep-forgetfulness. Otherwise, treat this as a +1 to the previous comment, in hope it gets fixed/implemented.

**Comment by lopsided98** [comment]
I have this same issue with my new M705 (M-R0073). Looking at the logs, it looks like Solaar is getting notifications about sleep and wakeup, but it doesn't handle them correctly.

**Comment by krrk** [comment]
This was fixed by #452. Note that it is required to start solaar with `--restart-on-wake-up` for the settings to persist on resume from suspend.

**Comment by ikelos** [comment]
I just ran into this, and the `--restart-on-wake-up` option isn't quickly findable.  Is there a reason it couldn't be enabled by default and/or discoverable within the UI?

**Comment by pfps** [comment]
I don't think that this option is needed for device settings to persist in the current version of Solaar.  If you are having problems with device settings not persisting please try with version 1.0.3 or the version from this repository.  If you are still having problems, please file a new bug report.

**Comment by ikelos** [comment]
I was on 1.0.3_rc1, but I'll give it a go on 1.0.3, and file a new issue if necessary.  Thanks!  5:)

**Comment by pfps** [comment]
What will probably be most useful is running Solaar with -dd (which produces tons of output) and then including the output from when your system resumed.  That will show what communications happened with the device.

**Comment by ikelos** [comment]
Ok, cool.  I'll give 1.0.3 a shot, and if it still goes wrong, I'll try to create the conditions with `-dd` and file a new bug with that information.

**Comment by a-lauz** [comment]
sorry to necro this closed issue but I'm on ver.  1.1.11+dfsg-2 for Linux Mint and I'm having this same problem with my K400 PLUS as others have mentioned, even after I added `--restart-on-wake-up` to the app's launch parameters.  every single time after just a few minutes of not touching anything, I have to open the application and play with the on/off toggles for it to remember my settings.  I just wish Logitech Options+ worked with Linux like it does in Windows.  LogiOpsGUI doesn't support keyboards yet and I've let the developer know about my problem, he's still working on it but it will take a long time to get anything.

**Comment by pfps** [comment]
restart-on-wake-up should not be needed.   

Version 1.1.11 is rather old - you should use version 1.1.14 or download from this repository.

If the problem persists, file a new bug report.
